### PR TITLE
Update to 1.71

### LINKED ISSUE #5528: [Chore]: upgrade Code to 1.71
https://github.com/microsoft/vscode/releases/tag/1.71.0 is out now.

### PR DESCRIPTION
**Pull request by fritterhoff**
As a first attempt the required fixes/changes for the code version 1.71.

 - The telemetry fix required some adjustments. I also included a fix that was recently merged upstream (https://github.com/microsoft/vscode/commit/b108bc8294ce920fcf2ee8d53f97c3bcf3316e1c) 
 - The CLI changes seem to be included in the upstream version now. Feel free to remove the remaining changes...

## Testing Plan

To test this locally:

1. run this:
```shell
gh pr checkout 5535 && git submodule update --init && quilt push -a && yarn && yarn watch --port 3000
```
2. navigate to [localhost:3000](http://localhost:3000)

### Patches tested

The following patches were tested locally:
- [x] base-path.diff
- [x] cli-window-open.diff
- [x] connection-type.diff
- [x] disable-builtin-ext-update.diff
- [x] disable-downloads.diff
- [x] exec-argv.diff
- [x] github-auth.diff
- [x] heartbeat.diff 
- [x] insecure-notification.diff
- [x] integration.diff
- [x] local-storage.diff
- [x] log-level.diff
- [x] logout.diff
- [x] marketplace.diff
- [x] parent-origin.diff
- [x] proposed-api.diff
- [x] proxy-uri.diff
- [x] safari.diff
- [x] service-worker.diff
- [x] sourcemaps.diff
- [x] store-socket.diff
- [x] telemetry.diff
- [x] unique-db.diff
- [x] update-check.diff
- [x] webview.diff

Fixes #5528

### CODE REVIEW

File: patches/telemetry.diff [hunk 1]
```diff
@@ -103,14 +138,6 @@ Index: code-server/lib/vscode/src/vs/workbench/services/telemetry/browser/teleme
...
 -				sendErrorTelemetry: this.sendErrorTelemetry,
 -			};
--			this.impl = this._register(new BaseTelemetryService(config, configurationService, productService));
--
--			if (getTelemetryLevel(configurationService) !== TelemetryLevel.NONE) {
--				// If we cannot fetch the endpoint it means it is down and we should not send any telemetry.
--				// This is most likely due to ad blockers
--				fetch(telemetryEndpointUrl, { method: 'POST' }).catch(err => {
--					this.impl = NullTelemetryService;
--				});
```
Comment (fritterhoff) [inline patches/telemetry.diff:113]: I actually have no idea what happened with this segment in the original version. I have not found any commit where this was removed or changed. Maybe sbd. finds the reason...

**Comment by fritterhoff** [comment]
@benz0li feel free to give the CI artificats a try. From my point of view it looks fine.

**Comment by benz0li** [comment]
@fritterhoff `code-server-4.6.1-linux-amd64.tar.gz` of the [artifacts](https://github.com/fritterhoff/code-server/actions/runs/2983562852) is deployed at https://vscode-r.jupyter.b-data.ch; Image `R (base:test) + code-server`.

Functionality [modified by patches] tested and found to work:

* base-path
* cli-window-open
* local-storage
    * I assume this is about settings
* marketplace
* proxy-url
* service-worker
* webview

👉 Same as https://github.com/coder/code-server/issues/5417#issuecomment-1211590250.

**Comment by benz0li** [comment]
@fritterhoff Could you please check if rendering is working properly for you at https://vscode-r.jupyter.b-data.ch?:

I get the following warning in the browser's web console:

```
[Log]  WARN – "Canvas could not be loaded. Falling back to the dom renderer type." (workbench.web.main.js, line 603)
TypeError: Type error
observe
(anonymous function) — DevicePixelObserver.ts:32
e — CanvasRenderer.ts:70
(anonymous function) — CanvasAddon.ts:32
(anonymous function) — AddonManager.ts:34
(anonymous function) — testingOutputPeek.ts:1023
asyncFunctionResume
(anonymous function)
promiseReactionJobWithoutPromise
promiseReactionJob
```

When I reload the page with an open Terminal, it does not get rendered properly.

**Comment by fritterhoff** [comment]
I'll check it asap.

**Comment by fritterhoff** [comment]
So I can at least point to the commit where this change was introduced: https://github.com/microsoft/vscode/commit/2f72682e5d0b9e2115b3062f55f47c90f134db7c

Have to debug that further 😉

**Comment by fritterhoff** [comment]
> I'll check it asap.
> 
> Well.. yes the rendering seems to fail. I'll try it with my own containers.

grafik

My own containers work fine so maybe there is a bug in your Dockerfile...

**Comment by fritterhoff** [comment]
> I'll check it asap.
> 
> Well.. yes the rendering seems to fail. I'll try it with my own containers.

So I tried again. No it works fine, beside some unusual spacing in the terminal and a rather slow loading. Also reloading works. I have no idea what actually failed yesterday. Maybe the browser only ran into a timeout or something like that.

**Comment by benz0li** [comment]
Everything works fine with Firefox as it uses the WebGL renderer.

Safari falls back to the dom render, as canvas can not be loaded [and WebGL is not supported].

## 1.71.0

### Screenshot



### Log

nothing about xterm

## 1.70.2

### Screenshot



### Log

```
[I 2022-09-04 09:21:24.192 SingleUserLabApp log:189] 200 GET /user/benz0li/code-server/stable-e4503b30fc78200f846c62cf8091b76ff5547662/static/node_modules/xterm/lib/xterm.js (benz0li@[redacted]) 51.26ms
[I 2022-09-04 09:21:24.318 SingleUserLabApp log:189] 200 GET /user/benz0li/code-server/stable-e4503b30fc78200f846c62cf8091b76ff5547662/static/node_modules/xterm-addon-unicode11/lib/xterm-addon-unicode11.js (benz0li@[redacted]) 11.81ms
[I 2022-09-04 09:21:24.427 SingleUserLabApp log:189] 200 GET /user/benz0li/code-server/stable-e4503b30fc78200f846c62cf8091b76ff5547662/static/node_modules/xterm-addon-unicode11/lib/xterm-addon-unicode11.js.map (benz0li@[redacted]) 10.74ms
[I 2022-09-04 09:21:24.539 SingleUserLabApp log:189] 200 GET /user/benz0li/code-server/stable-e4503b30fc78200f846c62cf8091b76ff5547662/static/node_modules/xterm/lib/xterm.js.map (benz0li@[redacted]) 189.38ms
```

**Comment by benz0li** [comment]
Workaround for Safari: Disable Gpu Acceleration

### Settings

safari_workaround

### Settings (JSON)

```json
"terminal.integrated.gpuAcceleration": "off"
```

**Comment by fritterhoff** [comment]
Damn it. Sadly I don't have the option to test safari. That will get complicated to reproduce/fix. Can you maybe test codespaces?

**Comment by fritterhoff** [comment]
I only made some quick checks using BrowserStack. Here Safari works fine but I'm not sure if they make any modifications or if I made did not test it enough. 

If i remember correct, @jsjoeio is also using a Mac. So maybe he can debug that a little bit more. I don't have access to a Macintosh device at all - sorry.

**Comment by benz0li** [comment]
### Safari: Web Console

browser-console

The link for `xterm-addon-canvas.js:0` points to https://vscode-r.jupyter.b-data.ch/user/benz0li/co…ules/xterm-addon-canvas/lib/xterm-addon-canvas.js  
ℹ️ There is a `…` in the URL and it is limited to 100 characters...

### Safari: Web Network

browser-network

👉 According to the network pane, `xterm-addon-canvas.js` is served from memory.

**Comment by benz0li** [comment]
> Can you maybe test codespaces?

No, unfortunately not. I am mainly using my self-hosted GitLab instance and a free account on both https://github.com and https://gitlab.com.

**Comment by benz0li** [comment]
> So I can at least point to the commit where this change was introduced: [microsoft/vscode@2f72682](https://github.com/microsoft/vscode/commit/2f72682e5d0b9e2115b3062f55f47c90f134db7c)

Well... xterm has been updated to v5.0.0 (beta 32): Many [issues](https://github.com/xtermjs/xterm.js/issues) with this version.

ℹ️ v1.70.2 used xterm v4.20.0 (beta 20).

**Comment by fritterhoff** [comment]
> > So I can at least point to the commit where this change was introduced: [microsoft/vscode@2f72682](https://github.com/microsoft/vscode/commit/2f72682e5d0b9e2115b3062f55f47c90f134db7c)
> 
> Well... xterm has been updated to v5.0.0 (beta 32): Many [issues](https://github.com/xtermjs/xterm.js/issues) with this version.
> 
> ℹ️ v1.70.2 used xterm v4.20.0 (beta 20).

I also saw the quite large amount of open issues but I'm not sure which actually are related to our problems. I only recognized that the upstream vscode project updated their dependency since the release of 1.71.

**Comment by benz0li** [comment]
> for safari, we don't enable webgl https://github.com/microsoft/vscode/blob/aa93bc5e86dc929f07c5bfa51771c43664d15bdd/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L1589

_Originally posted by @meganrogge in https://github.com/xtermjs/xterm.js/issues/3357#issuecomment-927986599_

**Comment by fritterhoff** [comment]
Out of curiosity. What Safari version do you use? According to https://caniuse.com/webgl2 webgl should be supported by the latest Safari versions.

**Comment by benz0li** [comment]
> Out of curiosity. What Safari version do you use?

Safari v15.6.1.

> According to https://caniuse.com/webgl2 webgl should be supported by the latest Safari versions.

Yes it is.

Disabled by default in VS Code: https://github.com/microsoft/vscode/blob/784b0177c56c607789f9638da7b6bf3230d47a8c/src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts#L285  
ℹ️ Link to codebase of v1.71.0

**Comment by fritterhoff** [comment]
Ah... I missed the edit in your comment yesterday.

**Comment by fritterhoff** [comment]
So I can at least confirm that the rendering of the terminal has a bug on iPads to, even though it is not the same as mentioned by you @benz0li.

**Comment by benz0li** [comment]
> [...], even though it is not the same as mentioned by you @benz0li.

It is the same issue. The Terminal looks like this after a reload:

1 71 0-reload

**Comment by benz0li** [comment]
Cross reference: https://github.com/microsoft/vscode/issues/160070

**Comment by jsjoeio** [comment]
Sorry, we had yesterday off for holiday in the US!

Can I just say this is such an awesome PR/conversation to come back to? Y'all are replacing me and @code-asher 😂

I have to tackle some other priorities but will get to this either today or tomorrow. And yes, I'm on macOS so I can test the Safari weirdness! Thank you two for getting the ball rolling!

File: patches/proxy-uri.diff [hunk 2]
```diff
@@ -35,31 +35,31 @@ Index: code-server/lib/vscode/src/vs/platform/remote/browser/remoteAuthorityReso
...
--		return { authority: { authority, host: host, port: port, connectionToken } };
-+		return { authority: { authority, host: host, port: port, connectionToken }, options };
- 	}
- 
- 	_clearResolvedAuthority(authority: string): void {
+-		const result: ResolverResult = { authority: { authority, host: host, port: port, connectionToken } };
++		const result: ResolverResult = { authority: { authority, host: host, port: port, connectionToken }, options };
+ 		RemoteAuthorities.set(authority, result.authority.host, result.authority.port);
+ 		this._cache.set(authority, result);
+ 		this._onDidChangeConnectionData.fire();
```
Comment (jsjoeio) [inline patches/proxy-uri.diff:62]: note to self: this change is fine, right?

File: patches/telemetry.diff [hunk 3]
```diff
@@ -1,5 +1,7 @@
 Add support for telemetry endpoint
 
+Contains some fixes included in https://github.com/microsoft/vscode/commit/b108bc8294ce920fcf2ee8d53f97c3bcf3316e1c
```
Comment (jsjoeio) [inline patches/telemetry.diff:3]: Nice addition 👍🏼

**Review by jsjoeio — COMMENTED** [review]
I think all of this looks fine. At least I don't see anything to be concerned about but I also have trouble reviewing these. I'll defer my review to @code-asher

**Comment by fritterhoff** [comment]
> ## Notes
> 
> Tested locally and things seem to be working 👍🏼
> 
> image
> 
> ### Safari 
> 
> Terminal works if I turn off GPU acceleration. 
> 
> image
> 
> 

Does it also work in case of enabled GPU acceleration?

**Comment by jsjoeio** [comment]
> Does it also work in case of enabled GPU acceleration?

I had it set to "auto" and it was not working. I had to make the change like @benz0li suggested.

**Comment by jsjoeio** [comment]
I spoke to @code-asher about this offline. His plate is full so I'm going to review this and help you merge this. I'm going to go through and test each patch now to make sure everything works as expected. I'll update the PR description too 👍🏼

**Comment by jsjoeio** [comment]
Heads-up: I'm halfway through testing this. I will finish the rest tomorrow. So far, everything is looking good though.

**Comment by fritterhoff** [comment]
Since upstream vs code "hot-fixed" the safari bug I would suggest to include this fix in code-server as well?

**Comment by fritterhoff** [comment]
@jsjoeio @benz0li I included the changes done by https://github.com/microsoft/vscode/pull/160332

**Comment by jsjoeio** [comment]
> For simplicity I placed it in a single diff file. That can be removed in case of a new release/update.

You rock. Thanks for doing that! I'm still reviewing this but should be done today.

**Comment by jsjoeio** [comment]
@fritterhoff I was talking to @code-asher about the `heartbeat` patch. Looks like upstream implemented it here: https://github.com/microsoft/vscode/pull/159239

Do you mind removing it in this PR? Otherwise I can do it in a followup PR.

**Comment by jsjoeio** [comment]
Safari fix works!

image

**Comment by jsjoeio** [comment]
I think the x86-64 job is timing out. You can increase this to 20m [here](https://github.com/coder/code-server/blob/main/.github/workflows/ci.yaml#L255)

**Review by jsjoeio — APPROVED** [review]
I tested everything and it all looks good! @fritterhoff feel free to mark ready and then I can merge

**Comment by jsjoeio** [comment]
Oops, fat finger click on the ready for review

**Comment by fritterhoff** [comment]
> I think the x86-64 job is timing out. You can increase this to 20m [here](https://github.com/coder/code-server/blob/main/.github/workflows/ci.yaml#L255)

Done

> @fritterhoff I was talking to @code-asher about the `heartbeat` patch. Looks like upstream implemented it here: [microsoft/vscode#159239](https://github.com/microsoft/vscode/pull/159239)
> 
> Do you mind removing it in this PR? Otherwise I can do it in a followup PR.

Would do that in a new PR and remove it in the next release?

**Comment by jsjoeio** [comment]
> Would do that in a new PR and remove it in the next release?

Oh! Good call. I thought it was in this version but looks like we need to wait for the next release.

**Review by jsjoeio — APPROVED** [review]
Woohoo!

**Comment by jsjoeio** [comment]
Opened an issue to track `heartbeat` removal: https://github.com/coder/code-server/issues/5544

**Comment by benz0li** [comment]
`code-server-4.6.1-1-linux-amd64.tar.gz` of the [artifacts](https://github.com/coder/code-server/actions/runs/3023906272) is deployed at https://vscode-r.jupyter.b-data.ch; Image `R (base:test) + code-server`.

**Comment by benz0li** [comment]
@jsjoeio The footprint of `code-server-4.6.1-1-linux-amd64` has increased to 626 MB. An increase of 264 MB compared to `code-server-4.6.1-linux-amd64`, whose footprint was 362 MB.

There is [grpc](https://www.npmjs.com/package/grpc) in `node_modules` of `code-server-4.6.1-1-linux-amd64` which alone is 236 MB in size.
👉 This package was not part of `code-server-4.6.1-linux-amd64`.

**Comment by jsjoeio** [comment]
That is a substantial increase. Thank you for pointing that out @benz0li! Let me and @code-asher dig in and see why that is.

**Comment by code-asher** [comment]
Hmm I am not sure; `yarn why grpc` does not seem to show anything.

**Comment by jsjoeio** [comment]
We figured it out using `npm-why`. Turns out `@coder/logger` -> `@google-cloud/logging` -> `grpc`

`@google-cloud/logging` is a `peerDependencies` and with our npm changes, it's being included in the build. We realized it's not being used so we're removing it from our logging library and bumping the version (https://github.com/coder/ts-logger/pull/16). We'll then update in code-server before releasing 4.7.0

**Pull request merged**